BinaryEncoding.md

```diff
@@ -234,6 +233,52 @@ This section must be preceded by a [Functions](#functions-section) section.
 | count | `varuint32` | count of entries to follow |
 | entries | `uint16*` | repeated indexes into the function table |
 
+### Function Names section
+
+This section may occur 0 or 1 times and does not change execution semantics. A
+validation error in this section is not reported and is treated as the section
+being absent. The expectation is that, when a binary WebAssembly module is
+viewed in a browser or other development environment, the names in this section
+will be used as the names of functions in the [text format](TextFormat.md).
+
+ID: `function_names`
+
+| Field | Type | Description |
+| ----- |  ----- | ----- |
+| names | `name*` | sequence of names |
+
+The number of names is determined by the number of declared functions. The
+sequence of names assigns a name to each function index. A name is:
+
+| Field | Type | Description |
+| ----- |  ----- | ----- |
+| size | varuint32 | string length, in bytes |
+| string | bytes | valid utf8 encoding |
```

Comment:
ASCII would be fine, even under the UTF8 requirement. :)  But I see your point. Can the spec somehow make UTF8 a "recommendation" or intention? It's weird to say it's a requirement when in fact none of the major consumers will check or otherwise care, nor are they expected to.